src/Connection.php

```diff
@@ -166,6 +167,12 @@ class Connection
 
     private SchemaManagerFactory $schemaManagerFactory;
 
+    private static bool $isChecking = false;
```

Comment:
these properties shouldn't be static - each connection object should have their own state